app/models/periodic_expense.rb

```diff
@@ -0,0 +1,50 @@
+class PeriodicExpense < ActiveRecord::Base
+  PERIODS = { 'weekly' => 1.week, 'monthly' => 1.month }
+
+  belongs_to :user
+
+  validates :name, :user, :amount, :period, :start_date, presence: true
+  validates :period, inclusion: { in: PERIODS.keys }
+  validate :end_date_is_after_start_date
+  validate :last_paid_at_is_between_start_and_end_dates
+
+  def period_as_time
+    PERIODS[period]
+  end
+
+  def ready_to_pay?
+    !next_pay.future?
```

Comment:
I prefer to follow the convention of having all dates suffixed with `_ast`, so it's more clear that they're dates. So this would become `next_pay_at`.
`next_pay` may sound like an instance of an actual payment

This is only a suggestion of course. I believe this isn't even in the guides